tidy3d/plugins/adjoint/components/geometry.py

```diff
@@ -0,0 +1,435 @@
+# pylint: disable=invalid-name
+"""Defines jax-compatible geometries and their conversion to grad monitors."""
+from __future__ import annotations
+
+from abc import ABC
+from typing import Tuple, Union
+
+import pydantic as pd
+import numpy as np
+import jax.numpy as jnp
+from jax.tree_util import register_pytree_node_class
+import jax
+
+from ....components.base import cached_property
+from ....components.types import Bound, Axis
+from ....components.geometry import Geometry, Box, PolySlab
+from ....components.data.monitor_data import FieldData, PermittivityData
+from ....components.monitor import FieldMonitor, PermittivityMonitor
+from ....constants import fp_eps, MICROMETER
+
+from .base import JaxObject
+from .types import JaxFloat, validate_jax_tuple, validate_jax_tuple_tuple
+from ..log import AdjointError
+
+# number of integration points per unit wavelength in material
+PTS_PER_WVL_INTEGRATION = 50
```

Comment:
Simply because the medium integration happens over the volume where the geometric integrations happen over all of the surfaces. So I wanted to use **fewer** points for the volume integration but keep the surface integration high enough to have accurate results without slowing things down.